### PR TITLE
DOC: Update documentation w.r.t. NPY_RELAXED_STRIDES_CHECKING

### PR DESCRIPTION
**Pull request by seberg**
This has been the default for a very long time, so this was
incorrect and mentioning any versions is probably not helpful.

### CODE REVIEW

File: doc/source/reference/arrays.ndarray.rst [hunk 1]
```diff
@@ -158,10 +158,11 @@ base offset itself is a multiple of `self.itemsize`. Understanding
 
 .. note::
 
-    Points (1) and (2) are not yet applied by default. Beginning with
-    NumPy 1.8.0, they are applied consistently only if the environment
-    variable ``NPY_RELAXED_STRIDES_CHECKING=1`` was defined when NumPy
-    was built. Eventually this will become the default.
```
Comment (eric-wieser) [inline doc/source/reference/arrays.ndarray.rst:164]: Can we include the version number when this default changed?

File: doc/source/reference/arrays.ndarray.rst [hunk 2]
```diff
@@ -158,10 +158,11 @@ base offset itself is a multiple of `self.itemsize`. Understanding
...
+    Points (1) and (2) can currently be disabled by the compile time
+    environmental variable ``NPY_RELAXED_STRIDES_CHECKING=0``.
```
Comment (seberg) [inline doc/source/reference/arrays.ndarray.rst:162]:
```suggestion
    environmental variable ``NPY_RELAXED_STRIDES_CHECKING=0``,
    which was the default before NumPy 1.10.
```

Comment (seberg) [reply doc/source/reference/arrays.ndarray.rst:162]: How far back do we want to actually do such bookkeeping? 1.10 is approaching 5 years ago.

**Review by mattip — APPROVED** [review]

**Comment by seberg** [comment]
Since Matti approved and it is a small change, I will just merge this. The old text is just wrong/outdated after all.

**Pull request merged**